cli/compose/convert/service.go

```diff
@@ -184,6 +192,70 @@ func Service(
 	return serviceSpec, nil
 }
 
+func convertAutoRange(oldAr composetypes.AutoRange) (swarm.AutoRange, error) {
+	if len(oldAr) <= 0 {
+		return swarm.AutoRange{}, nil
+	}
+
+	ar := make(swarm.AutoRange)
+	for k := range oldAr {
+		newR := strings.ToLower(k)
+		ar[newR] = make(map[string]string)
+		for sk, sv := range oldAr[k] {
+			if !CheckAutoRangeDeclaration(sv) {
+				return swarm.AutoRange{}, fmt.Errorf("Wrong parameter %s:%s for autoRange configuration", sk, sv)
+			}
+			ar[newR][sk] = sv
+		}
+	}
+	if err := CheckAutoRangeValues(ar); err != nil {
+		return swarm.AutoRange{}, err
+	}
+	return ar, nil
+}
+
+func CheckAutoRangeValues(ar swarm.AutoRange) error {
+	var min, max int = -1, -1
+	var threshold int
+
+	for category := range ar {
+		for k, v := range ar[category] {
+			switch k {
+			case "min":
+				min, _ = strconv.Atoi(v)
+			case "max":
+				max, _ = strconv.Atoi(v)
+			case "threshold":
+				threshold, _ = strconv.Atoi(v)
+				if threshold <= 0 || threshold > 100 {
+					return fmt.Errorf("Wrong vlaue for threshold")
+				}
+			default:
+				return fmt.Errorf("Unrecognized key %s for %s", k, category)
```

Comment:
nit: don't use upper case in error messages, so it can be nicely wrapped 👍
other nit: use `%q` instead of `%s`